engine/events/evolve_trade.asm

```diff
@@ -1,40 +1,19 @@
-EvolveTradeMon:
-; Verify the TradeMon's species name before
-; attempting to initiate a trade evolution.
-
-; The names of the trade evolutions in Blue (JP)
-; are checked. In that version, TradeMons that
-; can evolve are Graveler and Haunter.
-
-; In localization, this check was translated
-; before monster names were finalized.
-; Then, Haunter's name was "Spectre".
-; Since its name no longer starts with
-; "SP", it is prevented from evolving.
-
-; This may have been why Red/Green's trades
-; were used instead, where none can evolve.
-
-; This was fixed in Yellow.
-
+CheckEvolveTradeMonName: ; unused
```

Comment:
This is _not_ unused.

Reply:
_You are wrong._ Having no affect is not the same as being unused. This routine is absolutely executed under ordinary circumstances.

Reply:
> "Unused" does not imply "never executed"

Yes it does.

> If you want to be technical, then this is [dead code](https://en.wikipedia.org/wiki/Dead_code), but not [unreachable code](https://en.wikipedia.org/wiki/Unreachable_code).

You are absolutely right. That's exactly what the _original comments were there to communicate_. But you deleted all the comments in favor of a much more brief and unhelpful comment.

Reply:
> You are absolutely right.

Actually, I retract that. This routine is not dead code -- It contains some dead code.

Only Haunter/Spectre's trade evolution is broken. Graveler's trade evolution works just fine.

So again, this routine is absolutely positively not unused. It's bugged, as described by the original comments.

Reply:
I didn't say your new comments contain inaccuracies. I said they are more brief and less helpful.